src/libbson/src/bson/bson-json.c

```diff
@@ -894,6 +917,50 @@ _bson_json_parse_binary_elem (bson_json_reader_t *reader,
                               (int) vlen);
          }
       }
+   } else if (bs == BSON_JSON_LF_UUID) {
+      int nread = 0;
+      char uuid[33];
+
+      data->binary.has_binary = true;
+      data->binary.has_subtype = true;
+      data->binary.type = BSON_SUBTYPE_UUID;
+
+      /* Validate the UUID and extract relevant portions */
+      /* We can't use %x here as it allows +, -, and 0x prefixes */
+      SSCANF (val_w_null,
+              "%8[0-9a-fA-F]-%4[0-9a-fA-F]-%4[0-9a-fA-F]-%4[0-9a-fA-F]-%12[0-"
+              "9a-fA-F]%n",
```

Comment:
Using `sscanf` seemed like the most plausible way to check this. We don't check the return value here since we test that we read 36 bytes and encountered a null byte at the end below. This ensures that we don't accept strings that start out like a valid UUID but contain additional data in the end.

Reply:
Looking at the failures, I would suspect that not every system supports the `[0-9a-fA-F]` notation. I've changed it to use `%8c` instead but am getting access violation on Windows. The linux versions seem to work fine, so I'm at a bit of a loss as to what's causing this.

Reply:
Found it: the `sscanf_s` function used in Windows environments requires an additional argument specifying the buffer size when using `%c`.